src/hotspot/share/utilities/elfFile.cpp

```diff
@@ -1624,6 +1628,29 @@ bool DwarfFile::LineNumberProgram::get_filename_from_header(const uint32_t file_
   return false;
 }
 
+// Remove everything before the last slash including the slash itself to get the actual filename. This is required, for
+// example, for Clang debug builds which emit a relative path while GCC only emits the filename.
+void DwarfFile::LineNumberProgram::strip_path_prefix(char* filename, const size_t filename_len) {
+  char* last_slash = strrchr(filename, *os::file_separator());
+  if (last_slash != nullptr) {
+    uint16_t index_after_slash = (uint16_t)(last_slash + 1 - filename);
+    // Copy filename to beginning of buffer.
+    int bytes_written = jio_snprintf(filename, filename_len - index_after_slash, "%s", filename + index_after_slash);
```

Comment:
I don't think this is guaranteed to work since the memory areas you move may interleave. You should copy char-wise, or use `memmove(3)`.